cpp/src/arrow/extension/fixed_shape_tensor.h

```diff
@@ -74,10 +98,17 @@ class ARROW_EXPORT FixedShapeTensorType : public ExtensionType {
       const std::vector<int64_t>& permutation = {},
       const std::vector<std::string>& dim_names = {});
 
+  /// \brief Compute strides of FixedShapeTensorType
+  static Status ComputeStrides(const FixedWidthType& type,
```

Comment:
The reason it's like this now is because both `FixedShapeTensorType` and `FixedShapeTensorArray` are using `ComputeStrides`, but `ComputeStrides` is a `FixedShapeTensorType` method. I'll see if I can put it into another namespace.

Reply:
Moved to anonymous namespace.